### ISSUE TITLE
Add the ability to match a segment to an arc, similar to the "circle" button

### ISSUE THREAD

**Issue by afmenez** [openstreetmap/iD#2570]
it would be great to have the ability to fit a segment to an arc, similar to the existing "circle" button.  This would be useful in places like this http://www.openstreetmap.org/#map=17/-30.07305/-51.05239 or this http://www.openstreetmap.org/#map=16/-30.0503/-51.0119,which I had to tediously edit by hand and could not replicate the arc perfectly.

**Comment by bhousel** [comment]
Yes, agree... this is same as #2048

**Comment by afmenez** [comment]
I don´t agree it is the same issue.  The Circle button is for closed circles, I am talking about open arcs.  Unless you mean I will be able to use the circle button for arcs too.

**Comment by bhousel** [comment]
Oh you're right, fitting to an arc is different.

**Comment by pmailkeey** [comment]
it would be great to have the ability to fit a segment to an arc, similar to the existing "circle" button. This would be useful in places like this http://www.openstreetmap.org/#map=17/-30.07305/-51.05239 or this http://www.openstreetmap.org/#map=16/-30.0503/-51.0119 , which I had to tediously edit by hand and could not replicate the arc perfectly.

I'm not convinced the first one is arcworthy. The second one is, with a bit of effort - and I've circlified those with the circle function.

**Comment by afmenez** [comment]
Sorry, i don't understand you comment.  My suggestion is to add an "arc" button to reduce the effort, not to increase it.  Maybe I didn't express myself correctly, maybe a better example would be the "rectify" button, which works on segments.

**Comment by pmailkeey** [comment]
It is possible to tidy arcs up - by adding a temporary circle to the arc, using the circle function to make a smooth circle, then delete the circle to just leave the arc neat.

**Comment by afmenez** [comment]
Yes, for my second example above your method works, but for the first it may be harder than doing it by hand.
My point is not that it is impossible to do arcs currently.  I am just suggesting an easier way to do them.

**Comment by pmailkeey** [comment]
I think there might be plans for a 'smoothness' function - that works in several situations - making straight lines straight - without losing nodes - and curves smooth - even possibly ones that aren't a fixed radius.

**Comment by homersimpsons** [comment]
I would love this to :)

**Comment by kallejre** [comment]
Is this suggestion still being considered (for iD v3)? It's hybrid of straight line and full circle. In terms of UI, arc button could replace make-circular-button for unenclosed lines.

Forementioned #2048 mentions workaround by drawing circle full circle and then circlifing line. Sadly by using given tool, there is no way to guarantee, that endpoints of arc remain in planned positions. Even if determining radius of arc is too difficult to implement, manually adjusting arc radius is still better than nothing.

**Comment by homersimpsons** [comment]
As a workaround, I often draw a line that connected to the segment I want to be circular and I do circle the line which makes an arc.

Steps:
1. Simplify the future arc if possible (3 dot line is the best)
2. Draw a "circular" line connected to every dot
3. Round that line
4. Delete the line

![Example](https://user-images.githubusercontent.com/16977446/74107765-a3d21d80-4b73-11ea-8dab-aee10970a969.gif)

Side Notes:
I will Often keep only 3 dot on the line to go faster

**Comment by kaligrafy** [comment]
+1
